microsoft/testsuites/core/storage.py

```diff
@@ -242,6 +251,244 @@ def verify_os_partition_identifier(self, log: Logger, node: RemoteNode) -> None:
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case will verify that the premium ssd data disks disks can
+        be added serially while the vm is running. The test steps are same as
+        `hot_add_disk_serial`.
+        """,
+        requirement=simple_requirement(disk=DiskPremiumSSDLRS()),
+    )
+    def hot_add_disk_serial_premium_ssd(self, log: Logger, node: Node) -> None:
+        self._hot_add_disk_serial(
+            log, node, DiskType.PremiumSSDLRS, self.DEFAULT_DISK_SIZE_IN_GB
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case will verify that the standard HDD data disks can
+        be added in parallel while the vm is running.
+        Steps:
+        1. Get maximum number of data disk for the current vm_size.
+        2. Get the number of data disks already added to the vm.
+        3. Add maximum number of data disks to the VM in parallel.
+        4. Verify that the disks are added are available in the OS.
+        5. Remove the disks from the vm in parallel.
+        6. Verify that the disks are removed from the OS.
+        """,
+        requirement=simple_requirement(disk=DiskStandardHDDLRS()),
+    )
+    def hot_add_disk_parallel(self, log: Logger, node: Node) -> None:
+        self._hot_add_disk_parallel(
+            log, node, DiskType.StandardHDDLRS, self.DEFAULT_DISK_SIZE_IN_GB
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case will verify that the standard ssd data disks disks can
+        be added serially while the vm is running. The test steps are same as
+        `hot_add_disk_parallel`.
+        """,
+        requirement=simple_requirement(disk=DiskStandardSSDLRS()),
+    )
+    def hot_add_disk_parallel_standard_ssd(self, log: Logger, node: Node) -> None:
+        self._hot_add_disk_parallel(
+            log, node, DiskType.StandardSSDLRS, self.DEFAULT_DISK_SIZE_IN_GB
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case will verify that the premium ssd data disks disks can
+        be added serially while the vm is running. The test steps are same as
+        `hot_add_disk_parallel`.
+        """,
+        requirement=simple_requirement(disk=DiskPremiumSSDLRS()),
+    )
+    def hot_add_disk_parallel_premium_ssd(self, log: Logger, node: Node) -> None:
+        self._hot_add_disk_parallel(
+            log, node, DiskType.PremiumSSDLRS, self.DEFAULT_DISK_SIZE_IN_GB
+        )
+
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        self.added_data_disks: List[str] = []
+
+    def after_case(self, log: Logger, **kwargs: Any) -> None:
+        node: Node = kwargs["node"]
+        disk = node.features[Disk]
+
+        # cleanup any disks added as part of the test
+        # If the cleanup operation fails, mark node to be recycled
+        try:
+            disk.remove_data_disk(self.added_data_disks)
```

Comment:
Don't use the variables of `self`. The test cases may run in different nodes, there is conflict potentially. Because the disks names are decided in feature, so feature should be able to know which disk need to be removed. The `remove_data_disk` can accept empty arguments, it means remove all added disks.